### ISSUE TITLE
[BUG] In example training script gradients did not sync

### ISSUE THREAD

**Issue by ChenchaoZhao** [NVIDIA/Megatron-LM#1685]
**Describe the bug**
At line https://github.com/NVIDIA/Megatron-LM/blob/main/examples/run_simple_mcore_train_loop.py#L146

It did not run https://github.com/NVIDIA/Megatron-LM/blob/5653514e7c70a2adec71935d20c01ded5508c8b3/megatron/core/distributed/finalize_model_grads.py#L386

Thus, gradients were not synced.

**To Reproduce**
Steps to reproduce the behavior. The easier it is to reproduce the faster it will get maintainer attention.

Run https://github.com/NVIDIA/Megatron-LM/blob/main/examples/run_simple_mcore_train_loop.py

**Expected behavior**

Add sync gradient in the demo script.

**Stack trace/logs**
If applicable, add the stack trace or logs from the time of the error.

**Environment (please complete the following information):**
 - Megatron-LM commit ID
 - PyTorch version
 - CUDA version
 - NCCL version

**Proposed fix**
If you have a proposal for how to fix the issue state it here or link to a PR.

**Additional context**

**Comment by sbhavani** [comment]
You're right the example script was missing grad synchronization across distributed ranks.

We are adding the missing gradient synchronization with minimal changes:

1. Import `finalize_model_grads`
2. Add compatibility attributes for the model
3. Call `finalize_model_grads([gpt_model])` after `forward_backward_func()` and before `optim.step()`

We'll have a commit with the fix shortly. Thanks for helping improve the examples!

**Comment by sbhavani** [comment]
This issue has been fixed and merged to main in commit [feb68a3d9](https://github.com/NVIDIA/Megatron-LM/commit/feb68a3d9)
